pkg/file/buffer_test.go

```diff
@@ -39,24 +39,19 @@ func TestChunkPipe(t *testing.T) {
 			t.Parallel()
 
 			buf := file.NewChunkPipe()
-			sizeC := make(chan int, 255)
-			errC := make(chan error, 1)
+			sizeC := make(chan int, 16)
```

Comment:
Since the channels are now the same size and the `close` calls have been removed, I recommend using `result` type and having just one channel (similarly, as you have done below):

```Golang
// ...
type result struct {
	n   int
	err error
}
resultC := make(chan result, 16)
// ...
```

This will make the code easier to follow.

Reply:
This test case was structured differently and errC and sizeC are used separately. They are the same size but without any particular reason, except to provide extra buffer.

Reply:
What are the reasons for using them separately (both are the result of a single Read call)?

Reply:
Not sure, developer might have intended to use similar error handling structure in both tests?